spec/classes/profile/kubernetes_spec.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'nebula::profile::kubernetes' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:hiera_config) { 'spec/fixtures/hiera/kubernetes_config.yaml' }
+      let(:facts) { os_facts }
+
+      context 'with cluster unset' do
+        let(:params) { { cluster: nil } }
+
+        it { is_expected.not_to compile }
+      end
+
+      context 'with cluster set to one without a docker version' do
+        let(:params) { { cluster: 'implicit_docker_version' } }
+
+        it { is_expected.not_to compile }
+      end
+
+      context 'with cluster set to one without a kubernetes version' do
+        let(:params) { { cluster: 'implicit_kubernetes_version' } }
+
+        it { is_expected.not_to compile }
+      end
+
+      context 'with cluster set to first_cluster' do
+        let(:params) { { cluster: 'first_cluster' } }
+
+        %w[kubeadm kubelet].each do |package|
+          it { is_expected.to contain_package(package).with_ensure('1.14.2-00') }
+
+          [
+            'Class[Docker]',
+            'Apt::Source[kubernetes]',
+          ].each do |requirement|
+            it { is_expected.to contain_package(package).that_requires(requirement) }
+          end
+        end
+
+        it do
+          is_expected.to contain_file('/etc/systemd/system/docker.service.d')
+            .with_ensure('directory')
+        end
+
+        it do
+          is_expected.to contain_class('nebula::profile::docker').with(
+            version: '5:18.09.6~3-0~debian-stretch',
+          )
+        end
+
+        it do
+          is_expected.to contain_apt__source('kubernetes').with(
+            location: 'https://apt.kubernetes.io/',
+            release: 'kubernetes-xenial',
```

Comment:
Why xenial, given that kubernetes-stretch appears to exist there?

Reply:
Also, I think we're now leaning towards trying to minimize these tests that just duplicate static data from the profiles.

Reply:
That's a good question, and I think the answer is that I couldn't get stretch to work, although that was months ago, and maybe it just didn't exist at the time. I'll see what happens if I change it by hand on one of them, see if anything breaks.

Reply:
Confirmed, stretch doesn't actually offer kubelet or kubeadm (it only offers kubectl, which we don't need or want on the servers):

```
root@hatcher-k8s-work-002:~# apt-cache madison kubeadm
root@hatcher-k8s-work-002:~# apt-cache madison kubelet
root@hatcher-k8s-work-002:~# apt-cache madison kubectl
   kubectl |   1.6.0-00 | https://apt.kubernetes.io kubernetes-stretch/main amd64 Packages
   kubectl |   1.5.6-00 | https://apt.kubernetes.io kubernetes-stretch/main amd64 Packages
```

When I switch back to xenial, there are too many options to reasonably display here:

```
root@hatcher-k8s-work-002:~# apt-cache madison kubeadm | wc -l
110
root@hatcher-k8s-work-002:~# apt-cache madison kubelet | wc -l
151
```